src/resolvers/taxonomyResolvers.ts

```diff
@@ -0,0 +1,259 @@
+  GQLQueryNodeResourceArgs,
+  GQLQueryNodesArgs,
+  GQLTaxonomyContext,
+  GQLTaxonomyCrumb,
+  GQLTaxonomyEntity,
+  GQLWithArticle,
+} from "../types/schema";
+import { nodeToTaxonomyEntity } from "../utils/apiHelpers";
+import { filterMissingArticles, getArticleIdFromUrn, getLearningpathIdFromUrn } from "../utils/articleHelpers";
+
+export const Query = {
+  async node(
+    _: any,
+    { id, rootId, contextId }: GQLQueryNodeArgs,
+    context: ContextWithLoaders,
+  ): Promise<GQLNode | undefined> {
+    if (rootId) {
+      const children = await fetchChildren({ id: rootId, recursive: true }, context);
+      const node = children.find((child) => child.id === id);
+      if (node) return nodeToTaxonomyEntity(node, context.language);
+    }
+    if (id) {
+      const node = await context.loaders.nodeLoader.load({ id });
+      return nodeToTaxonomyEntity(node, context.language);
+    }
+    const nodes = await queryNodes({ contextId, includeContexts: true, filterProgrammes: false }, context);
+    if (nodes.length === 0) {
+      throw new Error(`No node found with contextId: ${contextId}`);
+    }
+    const entities = nodes.map((node) => nodeToTaxonomyEntity(node, context.language, contextId));
+    return entities.find((e) => e.contextId === contextId);
+  },
+  async nodes(
+    _: any,
+    { contentUri, filterVisible, metadataFilterKey, metadataFilterValue, ids }: GQLQueryNodesArgs,
+    context: ContextWithLoaders,
+  ): Promise<GQLNode[]> {
+    const nodes = await queryNodes(
+      {
+        contentURI: contentUri,
+        key: metadataFilterKey,
+        value: metadataFilterValue,
+        ids: ids,
+        includeContexts: true,
+        filterProgrammes: true,
+        language: context.language,
+      },
+      context,
+    );
+    const filtered = filterVisible ? nodes.filter((node) => node.contexts.find((context) => context.isVisible)) : nodes;
+    return filtered.map((node) => {
+      return nodeToTaxonomyEntity(node, context.language);
+    });
+  },
+  async nodeResource(
+    _: any,
+    { id, rootId, parentId }: GQLQueryNodeResourceArgs,
+    context: ContextWithLoaders,
+  ): Promise<GQLNode> {
+    const resource = await fetchNode({ id }, context);
+    const visibleCtx = resource.contexts
+      .filter((c) => c.isVisible)
+      .filter((c) => !c.rootId.startsWith("urn:programme"));
+    const subjectCtx = rootId ? visibleCtx.filter((c) => c.rootId === rootId) : visibleCtx;
+    const topicCtx = parentId ? subjectCtx.filter((c) => c.parentIds.includes(parentId)) : subjectCtx;
+
+    const path = topicCtx?.[0]?.path || resource.path;
+    const rank = topicCtx?.[0]?.rank;
+    const contextId = topicCtx?.[0]?.contextId;
+    const relevanceId = topicCtx?.[0]?.relevanceId || "urn:relevance:core";
+    const entity = nodeToTaxonomyEntity({ ...resource, contexts: visibleCtx }, context.language, contextId);
+    return { ...entity, contextId, path, rank, relevanceId };
+  },
+  async nodeCollection(
```

Comment:
Dette ble veldig abstrakt. Kanskje få inn en forklaring på hva en `nodeCollection` er? Spesielt når det eneste parameteret man kan sende inn er språk.

Reply:
Kun kopi av subjectCollection som blei lagt inn for å kunne returnere Node uten å knekke bruken. Kan legge inn kommentar.